e2e/cypress/integration/messaging/pinned_posts_spec.js

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @messaging
+
+describe('Messaging', () => {
+    let testTeam;
+    let testChannel;
+    let testUser;
+
+    before(() => {
+        cy.apiInitSetup().then(({team, channel, user}) => {
+            testUser = user;
+            testTeam = team;
+            testChannel = channel;
+        });
+    });
+
+    it('MM-T2167 Pin a post, view pinned posts', () => {
+        // # Login as the other user
+        cy.apiLogin(testUser);
+
+        // # Visit a test channel and post a message
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+        cy.postMessage('This is a post that is going to be pinned.');
+
+        cy.getLastPostId().then((postId) => {
+            // # On a message in center channel, click then pin the post to the channel
+            cy.getPostMenu(postId, 'Pin to Channel').click();
+
+            // * Find the 'Pinned' span in the post pre-header to verify that the post was actually pinned
+            cy.get(`#post_${postId}`).findByText('Pinned').should('exist');
```

Comment:
I'm all good, except for this one and will probably update the test case to something like,
`Message has Pinned badge in center but not in "Pinned Posts" RHS.`

With that, could you please add verification that the badge does not exist in RHS?